pkg/apis/serving/metadata_validation.go

```diff
@@ -25,6 +27,17 @@ import (
 // ValidateObjectMetadata validates that `metadata` stanza of the
 // resources is correct.
 func ValidateObjectMetadata(meta metav1.Object) *apis.FieldError {
-	return apis.ValidateObjectMetadata(meta).Also(
-		autoscaling.ValidateAnnotations(meta.GetAnnotations()).ViaField("annotations"))
+	return apis.ValidateObjectMetadata(meta).
+		Also(autoscaling.ValidateAnnotations(meta.GetAnnotations()).
+			Also(validateKnativeAnnotations(meta.GetAnnotations())).
+			ViaField("annotations"))
+}
+
+func validateKnativeAnnotations(annotations map[string]string) (errs *apis.FieldError) {
+	for key := range annotations {
+		if strings.HasPrefix(key, GroupName+"/") {
+			errs = errs.Also(apis.ErrInvalidKeyName(key, ""))
```

Comment:
Should we check `creator/lastModifier`?

Reply:
Yes #4763 has this logic so I did not want to duplicate the efforts. I will rebase this PR once 4763 is merged.